tdp/cli/commands/plan/edit.py

```diff
@@ -0,0 +1,215 @@
+            raise ValueError("Host is required after 'on' keyword.")
+    else:
+        host = None
+
+    # Get the extra vars and test if they are declared
+    if parsed_line.group(3) is not None:
+        extra_vars = parsed_line.group(3).split(" with ")[1]
+        if extra_vars == "":
+            raise ValueError("Extra vars are required after 'with' keyword.")
+        extra_vars = extra_vars.split(",")
+        extra_vars = [item.strip() for item in extra_vars]
+    else:
+        extra_vars = None
+
+    return (operation, host, extra_vars)
+
+
+@click.command(short_help="Edit Deployment plan.")
+@collections
+@database_dsn
+@vars
+def edit(
+    collections,
+    database_dsn,
+    vars,
+):
+    if not vars.exists():
+        raise click.BadParameter(f"{vars} does not exist.")
+    click.echo("Editing a deployment plan.")
+
+    program_exit = False
+
+    with get_session(database_dsn, commit_on_exit=True) as session:
+        planned_deployment_log = get_planned_deployment_log(session)
+        if planned_deployment_log is None:
+            raise click.ClickException("No planned deployment found.")
+
+        operations_hosts_var_names = [
+            operation.operation
+            + (" on " + operation.host if operation.host else "")
+            + (
+                " with " + ",".join(operation.extra_vars)
+                if operation.extra_vars
+                else ""
+            )
+            for operation in planned_deployment_log.operations
+        ]
+
+        temp_file = tempfile.NamedTemporaryFile(mode="w", suffix=".csv", delete=False)
+
+        header_text = f"""\
+# ------------------- DEPLOYMENT PLAN EDITOR --------------------------
+# Modify this file to edit the deployment plan.
+# Lines beginning with '#' are comments and will be ignored.
+#
+# Current plan id: {planned_deployment_log.id}
+# Temporary file: {temp_file.name}
+#
+# Execution Order:
+# - Operations are executed in the order they appear below.
+# - You can add, remove, or rearrange operations.
+# - Use `tdp nodes` command to view a list of available operations.
+#
+# Operations options:
+# - Specify a host with the `on` keyword.
+#   Example: hdfs_datanode_config on worker_02
+# - Add extra variables using the `with` keyword and separate with `,`.
+#   Example: wait_sleep with rolling-interval=5, mock-arg=example
+# ------------------- DEPLOYMENT PLAN EDITOR --------------------------
+"""
+
+        temp_file.write(header_text + "\n".join(operations_hosts_var_names))
+        temp_file.close()
+
+        click.echo(
+            f"Current plan id: {planned_deployment_log.id}\nTemporary file : {temp_file.name}"
+        )
+
+        # Remove empty elements and comments
+        # and get the operations, hosts and extra vars in a list
+        operations_hosts_var_names = [
+            _parse_line(line)
+            for line in operations_hosts_var_names
+            if line and not line.startswith("#")
+        ]
+
+        try:
```

Comment:
I'm not sure that this try is useful as it don't have any except.

Reply:
I use this try/finally block to remove the temporary file at the end of the program.
I will try your method ^^

Reply:
tested NOK ^^' the file is removed after the first read ^^'